sycamore/functions/rabin_karp.py

```diff
@@ -0,0 +1,88 @@
+from functools import reduce
+
+
+class RkHash:
+    """
+    Class RkHash encapsulates a 64-bit Rabin-Karp hash function.  It is
+    meant to hash one byte [0,255] at a time.  It's not the best hash
+    function, but has the nice property of allowing previous inputs to
+    be removed.  This makes it efficient for sliding-window applications.
+    """
+
+    __slots__ = ["shift", "prime", "primeShift", "val", "inv"]
```

Comment:
I don't see __slots__ used very often. Does it help here with performance?

Reply:
I just ran a benchmark and it's pretty much the same.  I'll take it out.